### PR TITLE
Harden exception report.  Fixes: #1365

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

Review and check off the below. Items that do not apply can also be checked off to indicate they
have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or open as a draft PR / add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Reviewed for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
Don -- please review and merge if it seems reasonable to you.  
Ran into this again today, and just wanted to take it off the table for v28.

File: appcontainer/ExceptionDialogModel.js [hunk 1]
```diff
@@ -58,16 +58,27 @@ export class ExceptionDialogModel {
...
-        await XH.exceptionHandler.logOnServerAsync({
+        const success = await XH.exceptionHandler.logOnServerAsync({
             exception,
             userMessage,
             userAlerted: true
         });
 
-        if (!options.requireReload) this.close();
+        if (success) {
+            await XH.alert({title: 'Message Sent', message: 'Your error has been reported.'});
```
Comment (febbraiod) [inline appcontainer/ExceptionDialogModel.js]: Are we sure we don't want to just toast here? Seems like the user has probably already done enough that we don't need to require another click. Realize thats not symmetrical with the code below, but the case there seems to justify the alert.

**Comment by lbwexler** [comment]
Yes, I thought about this as well, but was not sure about what kind of guarantees we could make about the health of our toast infrastructure in this situation (e.g. serious, early exception). 

 Figured since we were already depending on our dialog/alerting, best to avoid any additional dependencies on other parts of our infrastructure.....  Could go either way here....

**Comment by lbwexler** [comment]
Thinking about it some more, Don, I think you are correct -- can you make the change you are suggesting....  Good to avoid being annoying in this situation when our users might already be annoyed....

**Comment by lbwexler** [comment]
Don and I discussed at length and are going to move forward with the version as is.

**Pull request merged**